### PR TITLE
feat: event on group updates

### PR DESCRIPTION
**Pull request by solareon**
## Description

When updating groups for a player we don't trigger any event that can be consumed to update other resources. This adds a new event for resources to listen to and perform various functions such as updating UIs etc.

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

**Comment by FjamZoo** [comment]
Wouldnt it be better just to send the name and the grade that was modified, and the name that was removed?

That way scripts can do individual checks like if x was removed do y

instead of having to iterate over the entire table

File: server/player.lua [hunk 1]
```diff
@@ -294,6 +300,8 @@ local function removePlayerFromGang(citizenid, gangName)
 
     if not player.Offline then
         player.Functions.SetPlayerData('gangs', player.PlayerData.gangs)
+        TriggerEvent('QBCore:Server:OnGangsUpdate', player.PlayerData.source, gangName, nil)
+        TriggerClientEvent('QBCore:Client:OnGangsUpdate', player.PlayerData.source, gangName, nil)
```
Comment (FjamZoo) [inline server/player.lua:304]:
```suggestion
        TriggerEvent('QBCore:Server:OnGroupUpdate', player.PlayerData.source, gangName)
        TriggerClientEvent('QBCore:Client:OnGroupUpdate', player.PlayerData.source, gangName)
```

File: server/player.lua [hunk 2]
```diff
@@ -173,6 +175,8 @@ function RemovePlayerFromJob(citizenid, jobName)
 
     if not player.Offline then
         player.Functions.SetPlayerData('jobs', player.PlayerData.jobs)
+        TriggerEvent('QBCore:Server:OnJobsUpdate', player.PlayerData.source, jobName, nil)
+        TriggerClientEvent('QBCore:Client:OnJobsUpdate', player.PlayerData.source, jobName, nil)
```
Comment (FjamZoo) [inline server/player.lua:179]:
```suggestion
        TriggerEvent('QBCore:Server:OnGroupUpdate', player.PlayerData.source, jobName)
        TriggerClientEvent('QBCore:Client:OnGroupUpdate', player.PlayerData.source, jobName)
```

File: server/player.lua [hunk 3]
```diff
@@ -252,6 +256,8 @@ function AddPlayerToGang(citizenid, gangName, grade)
     if not player.Offline then
         player.PlayerData.gangs[gangName] = grade
         player.Functions.SetPlayerData('gangs', player.PlayerData.gangs)
+        TriggerEvent('QBCore:Server:OnGangsUpdate', player.PlayerData.source, gangName, grade)
+        TriggerClientEvent('QBCore:Client:OnGangsUpdate', player.PlayerData.source, gangName, grade)
```
Comment (FjamZoo) [inline server/player.lua:260]:
```suggestion
        TriggerEvent('QBCore:Server:OnGroupUpdate', player.PlayerData.source, gangName, grade)
        TriggerClientEvent('QBCore:Client:OnGroupUpdate', player.PlayerData.source, gangName, grade)
```

File: server/player.lua [hunk 4]
```diff
@@ -139,6 +139,8 @@ function AddPlayerToJob(citizenid, jobName, grade)
     if not player.Offline then
         player.PlayerData.jobs[jobName] = grade
         player.Functions.SetPlayerData('jobs', player.PlayerData.jobs)
+        TriggerEvent('QBCore:Server:OnJobsUpdate', player.PlayerData.source, jobName, grade)
+        TriggerClientEvent('QBCore:Client:OnJobsUpdate', player.PlayerData.source, jobName, grade)
```
Comment (FjamZoo) [inline server/player.lua:143]:
```suggestion
        TriggerEvent('QBCore:Server:OnGroupUpdate', player.PlayerData.source, jobName, grade)
        TriggerClientEvent('QBCore:Client:OnGroupUpdate', player.PlayerData.source, jobName, grade)
```

**Review by FjamZoo — CHANGES_REQUESTED** [review]

**Review by FjamZoo — APPROVED** [review]

**Review by TonybynMp4 — APPROVED** [review]

**Review by Manason — COMMENTED** [review]
Does this event already exist? If not shouldn't we name it qbx_core instead of QBCore?

**Pull request merged**